src/adapters/node.js

```diff
@@ -8,11 +7,20 @@
         defer: Q.defer,
         http: function(args) {
             var deff = Q.defer();
-            args.body = args.data;
+            if(args.data && typeof args.data === "string") {
+              // Setting args.json requires args.body to be a JSON as per request docs.
+              args.body = JSON.parse(args.data);
```

Comment:
Need need to use construction try catch for JSON.parse, string may is not JSON format

Reply:
Hmm, I would rather through parsing error to the caller than proceed further with the request. args.json is set to true. We are saying the input should be json...

Reply:
You can catch it and detailed describe.

Reply:
@mike1pol: I just realized that we are following the wrong thread. The PR #127 supersedes this. It is better to discuss all these issues there. If you agree I will close this PR?

Reply:
Ok